fairpy/items/saban_sethuraman_exchange_algorithm.py

```diff
@@ -0,0 +1,678 @@
+    :return:owner houses after this algo.
+
+
+    >>> PreferenceLists = {1: ['b', 'a'], 2: ['c', 'b'], 3: ['d', 'c'], 4: ['d']}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'a', 2: 'b', 3: 'c', 4: 'd'}
+
+    >>> PreferenceLists = {1: [{'a', 'b'}], 2: ['a', 'b']}
+    >>> house = {1: 'a', 2: 'b'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'b', 2: 'a'}
+
+    >>> PreferenceLists = {1: {'a': 3, 'b': 3}, 2: {'a': 5, 'b': 1}}
+    >>> house = {1: 'a', 2: 'b'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'b', 2: 'a'}
+
+    >>> PreferenceLists = {1: {'b': 3, 'a': 2}, 2: {'c': 5, 'b': 1}, 3:{'d':9, 'c':8},4:{'d':3}}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'a', 2: 'b', 3: 'c', 4: 'd'}
+
+    >>> PreferenceLists = {1: ['b', 'a'], 2: ['c', 'b'], 3: ['d', 'c'], 4: ['a']}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'b', 2: 'c', 3: 'd', 4: 'a'}
+
+    >>> PreferenceLists = {1: {'b': 3, 'a': 2}, 2: {'c': 5, 'b': 1}, 3:{'d':9, 'c':8},4:{'a':3}}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'b', 2: 'c', 3: 'd', 4: 'a'}
+
+    >>> PreferenceLists = {1: ['b'], 2: ['a'], 3: ['c']}
+    >>> house = {1: 'a', 2: 'b', 3: 'c'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'b', 2: 'a', 3: 'c'}
+
+    >>> PreferenceLists = {1: {'b': 3}, 2: {'a': 5}, 3:{'c':9}}
+    >>> house = {1: 'a', 2: 'b', 3: 'c'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'b', 2: 'a', 3: 'c'}
+
+    >>> PreferenceLists = {1: [{'g', 'c'}], 2: [{'f', 'g', 'd'}], 3: [{'b', 'e'}, 'c'], 4: ['e'], 5: ['d'], 6: ['b', 'f'], 7: ['a']}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd', 5: 'e', 6: 'f', 7: 'g'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'g', 2: 'f', 3: 'c', 4: 'e', 5: 'd', 6: 'b', 7: 'a'}
+
+    >>> PreferenceLists = {1: {'g': 3, 'c': 3}, 2: {'f': 2, 'g': 2, 'd': 2}, 3: {'b': 9, 'e':9, 'c': 3}, 4: {'e': 1}, 5: {'d': 5},6: {'b': 1, 'f': 0}, 7:{'a':3}}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd', 5: 'e', 6: 'f', 7: 'g'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'g', 2: 'f', 3: 'c', 4: 'e', 5: 'd', 6: 'b', 7: 'a'}
+
+    >>> PreferenceLists = {1: [{'a', 'c'}],2: [{'a', 'b', 'd'}], 3: [{'c', 'e'}], 4: ['c'], 5: [{'a', 'f'}], 6: ['b']}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd', 5: 'e', 6: 'f'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'a', 2: 'd', 3: 'e', 4: 'c', 5: 'f', 6: 'b'}
+
+    >>> PreferenceLists = {1: {'a': 3, 'c': 3}, 2: {'a': 2, 'b': 2, 'd': 2}, 3: {'c': 9, 'e':9}, 4: {'c': 1}, 5: {'a': 5, 'f': 5},6: {'b': 1}}
+    >>> house = {1: 'a', 2: 'b', 3: 'c', 4: 'd', 5: 'e', 6: 'f'}
+    >>> print_SCCs(house, PreferenceLists)
+    {1: 'a', 2: 'd', 3: 'e', 4: 'c', 5: 'f', 6: 'b'}
+
+
+
+    """
+
+
+
```

Comment:
יש להוסיף הערות בעזרת logger.info, כדי לעזור למשתמשים להבין את שלבי האלגוריתם.

Reply:
In the main function (top_trading_cycles_with_indifferences) i use logger.info .
I use it to show a list of preferences for all agents and list houses with their owners.
At the end of the function the logger print the owner of all houses after the redistribution.

Reply:
OK. But there should also be logs in the intermediate steps. 
You can use logger.info instead of the comments you wrote with "#" in the code.